forest/jasmine/mobmat2traj.py

```diff
@@ -1,570 +1,1327 @@
+
+    Returns:
+        delta_x, delta_y, delta_t: float,
+            adjusted displacement and time difference
+            between two points
+    """
+    if start_t + delta_t > end_t:
+        temp = delta_t
+        delta_t = end_t - start_t
+        delta_x = delta_x * delta_t / temp
+        delta_y = delta_y * delta_t / temp
+    return delta_x, delta_y, delta_t
+
+
+def calculate_position(
+    start_t: float, end_t: float, try_t: float,
+    start_delta: float, end_delta: float
+) -> float:
+    """This function calculates the position of a point
+        at a given time.
+
+    Args:
+        start_t: float, start time
+        end_t: float, end time
+        try_t: float, time to be calculated
+        start_delta, end_delta: float, displacement
+
+    Returns:
+        float, position at a given time
+    """
+
+    time_diff = end_t - start_t + 1e-5
+    time_part1 = try_t - start_t
+    time_part2 = end_t - try_t
+    res = time_part2 / time_diff * start_delta + \
+        time_part1 / time_diff * end_delta
+
+    return res
+
+
+def update_table(
+    imp_table: np.ndarray, list_elem: list
+) -> np.ndarray:
+    return np.vstack((imp_table, np.array(list_elem)))
+
+
+def forward_impute(
+    start_t: float, start_x: float, start_y: float,
+    end_t: float, end_x: float, end_y: float,
+    bv_set: np.ndarray, switch: int, num: int, pars: list,
+    flight_table: np.ndarray, linearity: float,
+    mis_row: np.ndarray, pause_table: np.ndarray,
+    imp_table: np.ndarray, start_s: int, method: str,
+    counter: int
+) -> Tuple[np.ndarray, Tuple[int, float, float, float], int]:
+    """
+    This function imputes a missing interval
+        from the start point to the end point.
+
+    Args:
+        start_t: float, start time
+        start_x, start_y: float, start position
+        end_t: float, end time
+        end_x, end_y: float, end position
+        bv_set: np.ndarray, output from BV_select()
+        switch: int, the number of binary variables to be generated
+        num: int, checks the top k similarities
+        pars: list, the parameters that are required
+         for the calculate_k1 function
+        flight_table: np.ndarray, output from create_tables()
+        linearity: float, controls the smoothness of a trajectory
+        mis_row: np.ndarray, a row of missing interval table
+        pause_table: np.ndarray, output from create_tables()
+        imp_table: np.ndarray, output from create_tables()
+        start_s: int, status of the start point
+        method: str, the method to be used for calculation,
+         should be either 'TL', 'GL', or 'GLC'
+        counter: int, number of imputed trajectories
+
+    Returns:
+        imp_table: updated imp_table
+        tuple of 4 elements:
+            start_s: int, status of the start point
+            start_t: float, start time
+            start_x, start_y: float, start position
+        counter: updated counter
+    """
+
+    I0 = indicate_flight(
```

Comment:
Variable in function should be lowercase. Also, a more descriptive name would make the code more readable.